src/views/results/components/ResultsList.tsx

```diff
@@ -13,20 +13,37 @@ export const ResultsList: FunctionComponent<Props> = (
 ): ReactElement<'div'> => {
   const { results } = props;
   const { t } = useTranslation();
-
   const [sortedBy, setSortedBy] = React.useState<string>('');
+  const [searchTerm, setSearchTerm] = React.useState<string>('');
+  const [searchResults, setSearchResults] = React.useState<any>(null);
```

Comment:
`any` tyypin käyttämistä kannattaa välttää jos vaan mahdollista. Sitä ei ole kokonaan kielletty, koska sitä on vanhastaan käytössä, mutta tavoitteena on poistaa se kaikkialta.

Tässä tapauksessa `searchResults` etsii sopivia `results` objekteja eli on results-array. Results on määritelty täällä: https://github.com/Archinowsk/konsti-client/blob/master/src/typings/result.typings.ts. Se on tyypiltään:

```
export interface Result {
  username: string;
  enteredGame: Signup;
}
```

Tiedostossa `ResultsList.tsx` importataan jo Results-tyyppi: https://github.com/Archinowsk/konsti-client/blob/search_field_to_all_results/src/views/results/components/ResultsList.tsx#L5

Tuon `searchResults` tilamuuttujan voi siis tyypittää näin: 

```  
const [searchResults, setSearchResults] = React.useState<Result[]>([]);
```

Reply:
Huomasin, että `results`-muuttujilla on myös ylimääräinen muuttuja `_id`. Jatkohommana voisi olla sen siivoaminen pois. Se luultavasti palautetaan kannasta, mikä ei ole tarkoituksenmukaista. Kannattaa kattoa, siivoaako sen pois suoraan, kun luetaan tietoja kannasta vai vasta frontissa, kun kutsutaan rajapintaa.